scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java

```diff
@@ -0,0 +1,80 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.persistence.NonUniqueResultException;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.NestedServletException;
+
+import edu.asu.diging.scriptoocloud.core.data.ProjectRepository;
+import edu.asu.diging.scriptoocloud.core.model.Project;
+import edu.asu.diging.scriptoocloud.core.model.impl.ProjectImpl;
+import edu.asu.diging.scriptoocloud.core.service.ProjectManager;
+import edu.asu.diging.simpleusers.core.model.IUser;
+
+@Transactional
+@Service
+public class ProjectManagerImpl implements ProjectManager {
+	
+	private final Logger logger=LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private ProjectRepository projectRepo;
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.scriptoocloud.core.service.impl.ProjectManager#createProject(java.lang.String, java.lang.String)
+     */
+	@Override
+    public Project createProject(String name, String description, IUser user) {
+        Project project = new ProjectImpl();
+        project.setName(name);
+        project.setUser(user);
+        project.setDescription(description);
+        project.setCreationDate(ZonedDateTime.now());
+        
+        return projectRepo.save((ProjectImpl)project);
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.scriptoocloud.core.service.impl.ProjectManager#deleteProject(java.lang.String)
+     */
+    @Override
+    public void deleteProject(long id) {
+   
+        try {
+        	// long is a primitive data type, so we must convert to an object data type Long to check for null. 
+        	Long checkId = id;
+        	if(checkId == null) {
+        		throw new nullIDException("Project does not exist.");
+        	} else {
+        		projectRepo.deleteById(id);
+        	}
+        } catch (nullIDException err) {
+        	logger.error("Project does not exist.");
+        }
```

Comment:
couple of things here:
- why do you assign `Long checkId = id`?
- there is no point in throwing an exception just to catch it right away. Throwing exceptions is to notify upper layers that something went wrong that can't be handled.
- Classes start with an uppercase letter (`nullIDException`)
- if you would catch an exception then it needs to be properly handled. Right now you're swallowing the exception. all that one would see in the logs would be "Project does not exist." No Stacktrace, no id, nothing. It wouldn't be useful at all to track down what went wrong.

Reply:
-checkID removed, had another idea in mind but is now removed. 
-Removed try/catch block in ProjectManagerImpl. I think my understanding here was not correct. Should now be caught appropriately in the RemoveProjectController. 
-Refactored nullIDException to NullIDException
-With correct catching of the block (if I fixed it correctly), should display "Project {id} does not exist".